app/controllers/bulk_court_dates_controller.rb

```diff
@@ -0,0 +1,46 @@
+class BulkCourtDatesController < ApplicationController
+  include CourtDateParams
+
+  before_action :require_organization!
+
+  def new
+    authorize :application, :admin_or_supervisor?
+
+    @court_date = CourtDate.new
+  end
+
+  def create
+    authorize :application, :admin_or_supervisor?
+
+    case_group_id = params[:court_date][:case_group_id]
+    if case_group_id.empty?
+      @court_date = CourtDate.new(court_date_params(nil))
+      @court_date.errors.add(:base, "Case group must be selected.")
+      render :new
+      return
+    end
+
+    case_group = current_organization.case_groups.find(case_group_id)
+
+    court_dates = case_group.casa_cases.map do |casa_case|
+      CourtDate.new(court_date_params(casa_case).merge(casa_case: casa_case))
+    end
+
+    court_date_with_error = nil
+    ActiveRecord::Base.transaction do
```

Comment:
Using a transaction to prevent a scenario where one court date succeeds and one fails. The user could be confused and try to create it again.

Reply:
excellent